packages/ember-application/lib/system/resolver.js

```diff
@@ -419,5 +421,31 @@ export default EmberObject.extend({
     }
 
     Logger.info(symbol, parsedName.fullName, padding, this.lookupDescription(parsedName.fullName));
+  },
+
+  /**
+   Used to iterate all items of a given type.
+
+   @method eachForType
+   @param {String} type the type to search for
+   @param {Function} callback the function to invoke when an item is found
+   @private
+   */
+  eachForType(type, callback) {
+    let namespace = get(this, 'namespace');
+    let suffix = classify(type);
+    let typeRegexp = new RegExp(`${suffix}$`);
+
+    let knownKeys = keys(namespace);
+    for (let index = 0, length = knownKeys.length; index < length; index++) {
+      let name = knownKeys[index];
+
+      if (typeRegexp.test(name)) {
+        let namePrefix = name.slice(0, suffix.length * -1);
+        let dasherizedName = dasherize(namePrefix);
```

Comment:
We actually need something that turns `App.FooBarHelper` (or `modulePrefix/helpers/foo-bar` and `modulePrefix/foo-bar/helper` in module-land) into `helper:foo-bar`.

I will make a method (to allow folks to have an escape hatch), but I think the global version will still need to use `dasherize`.